### ISSUE TITLE
SSLError during installation

### ISSUE THREAD

**Issue by ashwinivd** [jupyterhub/the-littlest-jupyterhub#604]
### Bug description
TLJH installation on VM Ubuntu 18.04 gives a SSL certificate error

#### Expected behaviour
I set the proxy as described in the troubleshooting guide and also the HTTPS certificate interception as described on the website

#### Actual behaviour
Intstallation just crashes at this point despite doing all of the above.

### Your personal set up
 Ubuntu 18.04 server on a VM with 4 GB RAM 

Error message is below:-

Checking if TLJH is already installed...
Setting up hub environment
Installed python & virtual environment
Set up hub virtual environment
Setting up TLJH installer...
Upgraded pip
Ran /opt/tljh/hub/bin/pip install --upgrade git+https://github.com/jupyterhub/the-littlest-jupyterhub.git with exit code 1
WARNING: The directory '/home/baadalvm/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you may want sudo's -H flag.
Collecting git+https://github.com/jupyterhub/the-littlest-jupyterhub.git
  Cloning https://github.com/jupyterhub/the-littlest-jupyterhub.git to /tmp/pip-req-build-f7c6xzbx
  Running command git clone -q https://github.com/jupyterhub/the-littlest-jupyterhub.git /tmp/pip-req-build-f7c6xzbx
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:852)'),)': /simple/backoff/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:852)'),)': /simple/backoff/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:852)'),)': /simple/backoff/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:852)'),)': /simple/backoff/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:852)'),)': /simple/backoff/
Could not fetch URL https://pypi.org/simple/backoff/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/backoff/ (Caused by SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:852)'),)) - skipping
ERROR: Could not find a version that satisfies the requirement backoff (from the-littlest-jupyterhub==0.1) (from versions: none)
ERROR: No matching distribution found for backoff (from the-littlest-jupyterhub==0.1)
Could not fetch URL https://pypi.org/simple/pip/: There was a problem confirming the ssl certificate: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /simple/pip/ (Caused by SSLError(SSLError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:852)'),)) - skipping

Traceback (most recent call last):
  File "<stdin>", line 183, in <module>
  File "<stdin>", line 168, in main
  File "<stdin>", line 56, in run_subprocess
subprocess.CalledProcessError: Command '['/opt/tljh/hub/bin/pip', 'install', '--upgrade', 'git+https://github.com/jupyterhub/the-littlest-jupyterhub.git']' returned non-zero exit status 1.

**Comment by shriharsh-a** [comment]
@ashwinivd Were you able to resolve this issue?

**Comment by ashwinivd** [comment]
No, it never really worked for me.

**Comment by ashwinivd** [comment]
I gave this another try and it worked successfully this time. I was working with a virtual machine with Ubuntu 18.04. Followed this sequence of steps.
 - Installed updates for the system with sudo apt-get update, followed by sudo apt-get upgrade
 - restarted the VM
 - ran the commands to install python3 and curl (sudo apt install python3 python3-dev curl git)
 - Pip is not installed by default on my Ubuntu VM, so also did a sudo apt install python-pip
 - Further, I also installed pip for Python3, by doing sudo apt-install python3-pip (I think this is the step that helped)
 - then ran the curl command as given in the documentation

Installation ran successfully after that!
